<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>SceneGraphWidget</class>
 <widget class="QWidget" name="SceneGraphWidget">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>400</width>
    <height>711</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Form</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout">
   <property name="margin">
    <number>0</number>
   </property>
   <item>
    <widget class="QWidget" name="headerWidget" native="true">
     <layout class="QHBoxLayout" name="horizontalLayout_2">
      <property name="spacing">
       <number>9</number>
      </property>
      <property name="margin">
       <number>0</number>
      </property>
      <item>
       <widget class="QWidget" name="headerLeftWidget" native="true">
        <layout class="QHBoxLayout" name="horizontalLayout_3">
         <property name="margin">
          <number>0</number>
         </property>
         <item>
          <widget class="QLabel" name="label1">
           <property name="text">
            <string>Task:</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLabel" name="nameLabel">
           <property name="text">
            <string>Name</string>
           </property>
          </widget>
         </item>
         <item>
          <spacer name="horizontalSpacer_2">
           <property name="orientation">
            <enum>Qt::Horizontal</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>0</width>
             <height>0</height>
            </size>
           </property>
          </spacer>
         </item>
        </layout>
       </widget>
      </item>
      <item>
       <widget class="QWidget" name="headerRightWidget" native="true">
        <layout class="QHBoxLayout" name="horizontalLayout_4">
         <property name="margin">
          <number>0</number>
         </property>
         <item>
          <widget class="QLabel" name="label2">
           <property name="text">
            <string>SimSpark:</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLabel" name="sparkLabel">
           <property name="text">
            <string>Spark</string>
           </property>
          </widget>
         </item>
         <item>
          <spacer name="horizontalSpacer_3">
           <property name="orientation">
            <enum>Qt::Horizontal</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>0</width>
             <height>0</height>
            </size>
           </property>
          </spacer>
         </item>
        </layout>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="SparkTreeView" name="sceneGraphTreeView">
     <property name="contextMenuPolicy">
      <enum>Qt::CustomContextMenu</enum>
     </property>
     <property name="whatsThis">
      <string>The scene graph of the selected task.

you can interact with nodes using the context menu or the property area below.</string>
     </property>
    </widget>
   </item>
  </layout>
 </widget>
 <customwidgets>
  <customwidget>
   <class>SparkTreeView</class>
   <extends>QTreeView</extends>
   <header>sparktreeview.h</header>
  </customwidget>
 </customwidgets>
 <resources>
  <include location="../../imageresources.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>sceneGraphTreeView</sender>
   <signal>customContextMenuRequested(QPoint)</signal>
   <receiver>SceneGraphWidget</receiver>
   <slot>showItemContextMenu(QPoint)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>330</x>
     <y>144</y>
    </hint>
    <hint type="destinationlabel">
     <x>403</x>
     <y>132</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>sceneGraphTreeView</sender>
   <signal>clicked(QModelIndex)</signal>
   <receiver>SceneGraphWidget</receiver>
   <slot>onItemClick(QModelIndex)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>363</x>
     <y>177</y>
    </hint>
    <hint type="destinationlabel">
     <x>407</x>
     <y>178</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>updateModelData()</slot>
  <slot>expandAll()</slot>
  <slot>collapseAll()</slot>
  <slot>showItemContextMenu(QPoint)</slot>
  <slot>onItemClick(QModelIndex)</slot>
 </slots>
</ui>
